engine/src/lib.rs

```diff
@@ -178,20 +178,20 @@ mod contract {
         let mut io = Runtime;
         let state = state::get_state(&io).sdk_unwrap();
```

Comment:
I see this line is unnecessary as there is an internal function `internal_get_upgrade_index()`. can you remove `state` variable?